web.rb

```diff
@@ -218,72 +233,79 @@ def create_customer
   }.to_json
 end
 
-# This endpoint responds to webhooks sent by Stripe. To use it, you'll need
-# to add its URL (https://{your-app-name}.herokuapp.com/stripe-webhook)
-# in the webhook settings section of the Dashboard.
-# https://dashboard.stripe.com/account/webhooks
-post '/stripe-webhook' do
-  json = JSON.parse(request.body.read)
+# ===== PaymentIntent Manual Confirmation 
+# See https://stripe.com/docs/payments/payment-intents/ios-manual
 
-  # Retrieving the event from Stripe guarantees its authenticity
-  event = Stripe::Event.retrieve(json["id"])
-  source = event.data.object
+# This endpoint is used by the mobile example apps to create and confirm a PaymentIntent 
+# using manual confirmation. 
+# https://stripe.com/docs/api/payment_intents/create
+# https://stripe.com/docs/api/payment_intents/confirm
+# A real implementation would include controls to prevent misuse
+post '/confirm_payment_intent' do
+  authenticate!
+  payload = params
+  if request.content_type.include? 'application/json' and params.empty?
+    payload = Sinatra::IndifferentHash[JSON.parse(request.body.read)]
+  end
 
-  # For sources that require additional user action from your customer
-  # (e.g. authorizing the payment with their bank), you should use webhooks
-  # to capture a PaymentIntent after the source becomes chargeable.
-  # For more information, see https://stripe.com/docs/sources#best-practices
-  WEBHOOK_CHARGE_CREATION_TYPES = ['bancontact', 'giropay', 'ideal', 'sofort', 'three_d_secure']
-  if event.type == 'source.chargeable' && WEBHOOK_CHARGE_CREATION_TYPES.include?(source.type)
-    begin
-      create_and_capture_payment_intent(
-        source.amount,
-        source.id,
-        nil,
-        ['card'],
-        source.metadata["customer"],
-        source.metadata,
-        source.currency,
-        nil,
-        nil
+  begin
+    if payload[:payment_intent_id]
+      # Confirm the PaymentIntent
+      payment_intent = Stripe::PaymentIntent.confirm(payload[:payment_intent_id], {:use_stripe_sdk => true})
+    elsif payload[:payment_method_id]
+      # Create and confirm the PaymentIntent
+      payment_intent = Stripe::PaymentIntent.create(
+        :amount => 1099, # A real implementation would calculate the amount based on e.g. an order id
```

Comment:
This could feel like a bug for someone trying out the Standard Integration app without reading the example-ios-backend source — can we instead keep a hash of items/prices and have the client send up a real list of items and quantities? It is more work, but we don't support *that* many items.

Reply:
(i will also volunteer to do this!)

Reply:
Yes please!

Reply:
Sounds good, I'll take care of it. I don't think it should block this patch.